exercicios/para-casa/sistema_de_alunas_por_floracanabrava.py

```diff
@@ -0,0 +1,157 @@
+
+    return qtd_faltas
+
+    
+def consultar_notas_aluna():
+    nome = input("Por favor, nos informe o nome da aluna: ")
+    sobrenome = input("Agora informe o sobrenome da aluna: ")
+
+    valores_notas = dataset[(nome, sobrenome)]["Notas"]
+    print(f"As três notas de {nome} {sobrenome} são {valores_notas[0]}, {valores_notas[1]}, {valores_notas[2]}.")
+
+    return valores_notas
+    
+def consultar_status_aprovacao():
+     nome = input("Por favor, nos informe o nome da aluna: ")
+     sobrenome = input("Agora informe o sobrenome da aluna: ")
+
+     notas = dataset[(nome, sobrenome)]["Notas"]
+     media = float(sum(notas)/len(notas))
+     qtd_faltas = dataset[(nome, sobrenome)]["Presença"].count(False)
+     percentual_de_faltas = (qtd_faltas / 5) * 100
+     nota_participacao = dataset[(nome, sobrenome)]["Participação"]
+     nota_corte = 6
+
+     if media < nota_corte:
+         print(f"A nota final da aluna {nome} {sobrenome} foi {media:.2f}, abaixo da {nota_corte}. A aluna está reprovada.")
+     elif percentual_de_faltas > 20:
+        print(f"A aluna {nome} {sobrenome} está reprovada por falta e sua média final foi {media:.2f}.")
+     elif nota_participacao <= 6:
+         print(f"A aluna {nome} {sobrenome} está reprovada por participação e sua média final foi {media:.2f}")
+     else:
+         print(f"A aluna {nome} {sobrenome} está aprovada e sua média final foi {media:.2f}")
+
+def obter_notas():
+    quantidade_notas = 3
+    notas = []
+
+    for contador in range(int(quantidade_notas)):
+        while True:
+            try:
+                entrada = float(input(f"Insira a nota #{contador + 1}: "))
+                if entrada < 0 or entrada > 10:
+                    print("Entrada inválida. Você deve inserir um número entre 0 e 10.")
+                else:
+                    notas.append(entrada)
+                    break
+            except ValueError:
+                print("Entrada inválida. Por favor, insira um número válido.")
+
+    return notas 
+
+def obter_presenca():
+    quantidade_aulas = 5
+    aulas = [] 
+
+    for contador in range(int(quantidade_aulas)):
+        while True:
+            entrada = input(f"Insira a presença da aula #{contador + 1}: ")
+            try:
+                presenca = eval(entrada)
+                if isinstance(presenca, bool):
+                    aulas.append(presenca)
+                    break 
+                else:
+                    print("Entrada inválida. Por favor utilize True para presente e False para ausente.")
+            except NameError:
+                print("Entrada inválida. Por favor, insira True ou False.")
+
+    return aulas 
+   
+
+def salvar_dados_alunas(nome, sobrenome, turma, notas, presenca, participacao):
+    chave = (nome, sobrenome)
+    dataset[chave] = {
+        (nome, sobrenome):{
+           "Turma": turma,
+           "Notas": notas,
+           "Presença": presenca,
+           "Participação": participacao
+    },}
```

Comment:
A função salvar_dados_alunas está com uma estrutura de dados aninhado diferente da estrutura original do nosso dataset. Como já temos uma tupla na chave, não necessitava de outra tupla no corpo do dicionário. Quando realizamos um teste de consulta de notas com uma aluna cadastrada manualmente, podemos reparar que acaba dando erro por causa da implementação desta função.

**Sugestão:**
```suggestion
def salvar_dados_aluna(nome, sobrenome, turma, notas, presenca, participacao):
    chave = (nome, sobrenome)
    dataset[chave] = {
        "Turma": turma,
        "Notas": notas,
        "Presença": presenca,
        "Participação": participacao
    }
```